neonvm-runner/cmd/httpserver.go

```diff
@@ -87,7 +88,7 @@ func handleCPUChange(
 	logger *zap.Logger,
 	w http.ResponseWriter,
 	r *http.Request,
-	set func(*zap.Logger, vmv1.MilliCPU) error,
+	set func(*zap.Logger, vmv1.MilliCPU) (*int, error),
```

Comment:
what is the reason to use pointer for a error code?

Reply:
The intent was to help avoid needing to write the HTTP status in more places (so if `err != nil` then `nil` is converted to HTTP 500, and if `err != nil` then the status wouldn't be used anyways)

I'm not totally sure it's worth it though TBH

Reply:
>The intent was to help avoid needing to write the HTTP status in more places

Not sure I follow how it helps :( 

What can't be done with int but can be done with *int in that scenario, assuming we use default value 0 as a logical nil?

Reply:
It can be done either way, I just thought the semantics of having zero as a special value was a little odd :sweat_smile:

Reply:
In general in go it is a common practice to use default values (I guess mostly to avoid putting more pressure on GC and mem fragmentation, both are not related to our scenario) so at least to me that code was really confusing to read. In general I'd rather stick with common practices even if I don't like them, just to make cognitive load during reading less. Anyway, no hard feelings here.

Reply:
Makes sense - implemented in 4c35c85f898fc68b17f7c539c272b2ffe0f3d2bb